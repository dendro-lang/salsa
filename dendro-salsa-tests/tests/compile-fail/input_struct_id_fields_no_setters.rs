
#[salsa::jar(db = Db)]
struct Jar(MyInput);

trait Db: salsa::DbWithJar<Jar> {}

#[salsa::input(jar = Jar)]
struct MyInput {
    field: u32,
    #[id]
    id_one: u32,
}

#[salsa::db(Jar)]
#[derive(Default)]
struct Database {
    storage: salsa::Storage<Self>,
}

impl salsa::Database for Database {}

impl Db for Database {}


fn main() {
    let mut db = Database::default();
    let input = MyInput::new(&mut db, 3, 4);
    // should not compile as `id_one` should not have a setter
    // compile error: method `set_id_one` not found in scope
    input.set_id_one(1);
}