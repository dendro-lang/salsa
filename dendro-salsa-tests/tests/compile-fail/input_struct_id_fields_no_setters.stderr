error[E0599]: no method named `set_id_one` found for struct `MyInput` in the current scope
  --> tests/compile-fail/input_struct_id_fields_no_setters.rs:30:11
   |
7  | #[salsa::input(jar = Jar)]
   | -------------------------- method `set_id_one` not found for this struct
...
30 |     input.set_id_one(1);
   |           ^^^^^^^^^^ help: there is a method with a similar name: `id_one`
